lake/api/remote.go

```diff
@@ -12,42 +12,47 @@ import (
 	"github.com/brimdata/zed/lake/index"
 	"github.com/brimdata/zed/lakeparse"
 	"github.com/brimdata/zed/order"
+	"github.com/brimdata/zed/pkg/storage"
 	"github.com/brimdata/zed/zbuf"
 	"github.com/brimdata/zed/zio"
 	"github.com/brimdata/zed/zio/zngio"
 	"github.com/segmentio/ksuid"
 )
 
-type RemoteSession struct {
+type remote struct {
 	conn *client.Connection
 }
 
-var _ Interface = (*RemoteSession)(nil)
+var _ Interface = (*remote)(nil)
 
-func OpenRemoteLake(ctx context.Context, url string) (*RemoteSession, error) {
-	return &RemoteSession{
+func OpenRemoteLake(ctx context.Context, url string) (Interface, error) {
+	return &remote{
 		conn: client.NewConnectionTo(url),
 	}, nil
 }
 
-func NewRemoteWithConnection(conn *client.Connection) *RemoteSession {
-	return &RemoteSession{conn}
+func OpenRemoteLakeWithURI(ctx context.Context, uri *storage.URI) (Interface, error) {
+	return OpenRemoteLake(ctx, uri.String())
 }
 
-func (r *RemoteSession) PoolID(ctx context.Context, poolName string) (ksuid.KSUID, error) {
+func NewRemoteWithConnection(conn *client.Connection) *remote {
+	return &remote{conn}
+}
```

Comment:
1. The context parameters and error return parameters of `OpenRemoteLake` and `OpenRemoteLakeWithURI` are unused.
2. The name of `NewRemoteWithConnection` and its `*remote` return type are inconsistent with the other two constructors.
3. Three constructors feels pretty busy.

Based on all that, I think we should just have this:
```go
func NewRemoteLake(conn *client.Connection) Interface {
	return &remote{conn}
}
```